src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -98,6 +103,19 @@ case class Template(r1: Option[SamRecord],
     Template(x1, x2)
   }
 
+  /** Fixes mate information and sets mate cigar on all primary and supplementary (but not secondary) records. */
+  def fixMateInfo(): Unit = {
+    for (primary <- r1; supp <- r2Supplementals) {
+      SamPairUtil.setMateInformationOnSupplementalAlignment(supp.asSam, primary.asSam, true)
+    }
+    for (primary <- r2; supp <- r1Supplementals) {
+      SamPairUtil.setMateInformationOnSupplementalAlignment(supp.asSam, primary.asSam, true)
+    }
+    for (first <- r1; second <- r2) {
+      SamPairUtil.setMateInfo(first.asSam, second.asSam)
```

Comment:
Do you want to set the mate cigar?  I think if not given, the default is `false`
```suggestion
      SamPairUtil.setMateInfo(first.asSam, second.asSam, true)
```